cuenimby/cuenimby/tray.py

```
@@ -306,35 +312,24 @@ def _open_config(self) -> None:
                 editor = (os.environ.get("VISUAL")
                           or os.environ.get("EDITOR")
-                          or shutil.which("xdg-open"))
-                if editor and editor != shutil.which("xdg-open"):
-                    # Launch editor in a terminal so the user can interact
+                          or shutil.which("gedit")
+                          or shutil.which("nano")
+                          or "vi")
+                gui_editors = ("gedit", "kate", "mousepad", "xed")
+                if any(editor.endswith(g) for g in gui_editors):
+                    # GUI editors launch directly, no terminal needed
+                    subprocess.Popen([editor, config_path])
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)